src/common/settings.ts

```diff
@@ -39,6 +41,7 @@ type Format = {
 
 export interface ISettings {
   nativeServer: boolean;
+  languageServer: LanguageServer;
```

Comment:
Do we have plan to introduce new type of server in future? otherwise, I'd recommend using one setting and avoid making more scenarios that settings maybe not worked as users expected when modified together.

How about `nativeServer` setting with values of `auto | on | off` which is closer to alternative vscode's settings? we can keep it backward compatible while introducing new values.
I think when `nativeServer` chosen at https://github.com/astral-sh/ruff-vscode/pull/466#issuecomment-2122463407, we also realized that it could have values expanded in future.

Reply:
> How about `nativeServer` setting with values of `auto | on | off` which is closer to alternative vscode's settings?

This sounds like a pretty reasonable option to me. We can keep the "boolean" type for backward compatibility and map `true`/`false` to `on`/`off` respectively.

Reply:
Good idea @T-256! I'll rework this setting.